client/client_test.go

```diff
@@ -99,22 +93,59 @@ func Test_GetFlyteHealthCheckURL_ShouldReturnErrorWhenItCannotGetHealthCheckURLF
 }
 
 func Test_TakeAction_ShouldReturnSpecificErrorTypeAndMessageWhenResourceIsNotFound(t *testing.T) {
-	handler := func(w http.ResponseWriter, r *http.Request) { w.WriteHeader(http.StatusNotFound) }
-	server := httptest.NewServer(http.HandlerFunc(handler))
-	defer server.Close()
+	ts := mockServer(http.StatusNotFound, "")
+	defer ts.Close()
 
-	u, _ := url.Parse(server.URL + "/take/action/url")
+	c := newTestClient(ts.URL, t)
+	u, err := url.Parse(ts.URL + "/take/action/url")
+	require.NoError(t, err)
 
-	c := &client{takeActionURL: u, httpClient: &http.Client{
-		Timeout: 5 * time.Second,
-	}}
+	c.takeActionURL = u
+	_, err = c.TakeAction()
 
-	_, e := c.TakeAction()
+	require.IsType(t, NotFoundError{}, err)
+	assert.EqualError(t, err, fmt.Sprintf("Resource not found at %s/take/action/url", ts.URL))
+}
 
-	if assert.Error(t, e) {
-		assert.IsType(t, NotFoundError{}, e)
-		assert.Equal(t, fmt.Sprintf("Resource not found at %s/take/action/url", server.URL), e.Error())
-	}
+//should register pack with API, and have populated links
+func TestClient_CreatePack(t *testing.T) {
+
+	ts, rec := mockServerWithRecorder(http.StatusCreated, slackPackResponse)
+	defer ts.Close()
+
+	c := newTestClient(ts.URL, t)
+
+	err := c.CreatePack(Pack{Name: "Slack"})
+	require.NoError(t, err)
+
+	assert.NotNil(t, c.takeActionURL)
+	assert.Equal(t, "http://example.com/v1/packs/Slack/actions/take", c.takeActionURL.String())
+	assert.Len(t, rec.reqs, 1)
+
+	assert.NotNil(t, c.eventsURL)
+	assert.Equal(t, "http://example.com/v1/packs/Slack/events", c.eventsURL.String())
+	assert.Len(t, rec.reqs, 1)
+
+}
+
+func TestCreatePackShouldReturnErrorIfTakeActionsLinksAreNotSet(t *testing.T) {
```

Comment:
Can you make the test names a bit more consistent? i.e.  put underscores in like the above tests 'Test_CreatePack_ShouldReturnErrorIfTakeActionsLinksAreNotSet'